document RETURN NEXT and RETURN QUERY (#19589)
cockroachdb/docs: src/current/v25.2/plpgsql.md

#### `RETURN`

> Add a `RETURN` statement to a routine with an `OUT` parameter, `RETURNS VOID` clause, or `RETURNS SETOF` clause to exit the routine immediately.

Reviewer: nit
```suggestion
Add a `RETURN` statement to immediately exit a routine that includes an `OUT` parameter, `RETURNS VOID` clause, or `RETURNS SETOF` clause.
```

Author: Going to keep the original for consistency with the other sections!

~~~ sql
BEGIN
	...
	RETURN;
~~~

Add a `RETURN` statement in a scalar-returning function to return the result of an expression.

> In the following example, `RETURN` returns the square of the input argument.

Reviewer: nit
```suggestion
The following example uses `RETURN` to return the square of the input argument.
```

{% include_cached copy-clipboard.html %}
~~~ sql
CREATE FUNCTION square(x INT) RETURNS INT AS $$
BEGIN
	RETURN x * x;
END;
$$ LANGUAGE PLpgSQL;
~~~

#### `RETURN NEXT` and `RETURN QUERY`

Add `RETURN NEXT` or `RETURN QUERY` statements to a [set-returning function]({% link {{ page.version.version }}/create-function.md %}#create-a-function-that-returns-a-set-of-results) to append rows to the result set. `RETURN NEXT` and `RETURN QUERY` statements can be combined in a single function to build the result set.

Use `RETURN NEXT` within a set-returning function to append a row to the result set.

In the following example, `RETURN NEXT` returns a new row during each loop iteration.

{% include_cached copy-clipboard.html %}
~~~ sql
CREATE FUNCTION get_numbers() RETURNS SETOF INT AS $$
DECLARE
	i INT := 1;
BEGIN
	WHILE i <= 5 LOOP
		RETURN NEXT i;
		i := i + 1;
	END LOOP;
END
$$ LANGUAGE PLpgSQL;
~~~

> Use `RETURN QUERY` within a set-returning function to append the results of a SQL query to the result set.

Author: I believe it can, such as `VALUES` -- I think "results of a SQL query" implies that the SQL query produces a result.

	RETURN;
~~~

> #### `RETURN NEXT` and `RETURN QUERY`

Reviewer 2: We might want to add something to the `RETURN` section above about set-returning functions, since `RETURN` with no parameters serves a similar function there. Are we also missing some docs about using `RETURN` with parameters for scalar functions?

Author: Thanks! I went ahead and documented all this. Could you PTAL?
